src/airflow_tools/providers/http_to_data_lake/operators/http_to_data_lake.py

```diff
@@ -174,8 +176,12 @@ def _file_name(self, n_part) -> str:
         return file_name
 
     def _response_filter(self, response) -> BytesIO:
-        if not self.jmespath_expression:
+        if self.save_format in self.text_response_save_format:
```

Comment:
I think it's easier if instead of maintaining a list of text_response_save_format we just check if it's a JSON or JSONL, which will be less error prone when we add a new save_format and don't need to remember to add it to this list. I think here we just need:

- If there's a jmespath and save_format is JSON or JSONL -> apply JMESPATH
- If there's a jmespath and save_format is not JSON or JSONL -> error
- If save_format is JSON or JSONL -> response.json()
- Else -> response.text

Reply:
solved